MekHQ/src/mekhq/campaign/work/IPartWork.java

```diff
@@ -85,39 +85,41 @@ default boolean canChangeWorkMode() {
      * @return A string containing details regarding the part.
      */
     String getDetails(boolean includeRepairDetails);
+
     int getLocation();
-    
+
     Unit getUnit();
-    
+
     boolean isSalvaging();
-    
+
     String checkFixable();
-    
+
     void reservePart();
+
     void cancelReservation();
+
     boolean isBeingWorkedOn();
-    
-    int getMassRepairOptionType();
-    int getRepairPartType();
-    
-    public static int findCorrectMassRepairType(IPartWork part) {
-        if (part instanceof EquipmentPart && ((EquipmentPart)part).getType() instanceof WeaponType) {
-            return Part.REPAIR_PART_TYPE.WEAPON;
-        } else {            
+
+    PartRepairType getMassRepairOptionType();
+
+    PartRepairType getRepairPartType();
+
+    static PartRepairType findCorrectMassRepairType(IPartWork part) {
+        if ((part instanceof EquipmentPart) && (((EquipmentPart) part).getType() instanceof WeaponType)) {
+            return PartRepairType.WEAPON;
+        } else {
             return part.getMassRepairOptionType();
         }
     }
-    
-    public static int findCorrectRepairType(IPartWork part) {
-        if ((part instanceof EquipmentPart && ((EquipmentPart)part).getType() instanceof WeaponType) ||
-                (part instanceof MissingEquipmentPart && ((MissingEquipmentPart)part).getType() instanceof WeaponType)) {
-            return Part.REPAIR_PART_TYPE.WEAPON;
+
+    static PartRepairType findCorrectRepairType(IPartWork part) {
+        if (((part instanceof EquipmentPart) && (((EquipmentPart) part).getType() instanceof WeaponType))
+                || ((part instanceof MissingEquipmentPart) && (((MissingEquipmentPart) part).getType() instanceof WeaponType))) {
+            return PartRepairType.WEAPON;
+        } else if ((part instanceof EquipmentPart) && (((EquipmentPart) part).getType().hasFlag(MiscType.F_CLUB))) {
```

Comment:
Do all physical weapons have a club flag? Not that this PR will make it any worse, just curious.

Reply:
No idea, but I'm actually not sure that is properly implemented as it is. However, given the overall PR size, that look can be left until later.